politeiawww/user.go

```diff
@@ -200,6 +205,56 @@ func validatePubkey(publicKey string) ([]byte, error) {
 	return pk, nil
 }
 
+// formatName normalizes a contractor name to lowercase without leading and
+// trailing spaces.
+func formatName(name string) string {
+	return strings.ToLower(strings.TrimSpace(name))
+}
+
+func validateName(name string) error {
+	if len(name) < www.PolicyMinUsernameLength ||
+		len(name) > www.PolicyMaxUsernameLength {
+		log.Tracef("Name not within bounds: %s", name)
+		return www.UserError{
+			ErrorCode: www.ErrorStatusMalformedUsername,
+		}
+	}
+
+	if !validName.MatchString(name) {
+		log.Tracef("Name not valid: %s %s", name, validName.String())
+		return www.UserError{
+			ErrorCode: www.ErrorStatusMalformedUsername,
+		}
+	}
+
+	return nil
+}
+
+// formatLocation normalizes a contractor location to lowercase without leading and
+// trailing spaces.
+func formatLocation(location string) string {
+	return strings.ToLower(strings.TrimSpace(location))
+}
+
+func validateLocation(location string) error {
```

Comment:
The location regex doesn't accept `dallas, tx, usa`.

Also, shouldn't these be `MalformedLocation` errors?